PetAdoptions/cdk/pet_stack/resources/destroy_stack.sh

```diff
@@ -4,11 +4,168 @@
+        echo "🗑️  Force deleting VPC and related resources: $vpc"
+
+        # Delete NAT Gateways
+        local nat_gateways=$(aws ec2 describe-nat-gateways --filter "Name=vpc-id,Values=$vpc" --query 'NatGateways[?State==`available`].NatGatewayId' --output text)
+        for nat in $nat_gateways; do
+            echo "   Deleting NAT Gateway: $nat"
+            aws ec2 delete-nat-gateway --nat-gateway-id "$nat"
+        done
+
+        # Wait for NAT Gateways to be deleted
+        for nat in $nat_gateways; do
+            echo "   Waiting for NAT Gateway $nat to be deleted..."
+            aws ec2 wait nat-gateway-deleted --nat-gateway-ids "$nat" 2>/dev/null || true
+        done
+
+        # Delete Internet Gateway
+        local igws=$(aws ec2 describe-internet-gateways --filters "Name=attachment.vpc-id,Values=$vpc" --query 'InternetGateways[].InternetGatewayId' --output text)
+        for igw in $igws; do
+            echo "   Detaching and deleting Internet Gateway: $igw"
+            aws ec2 detach-internet-gateway --internet-gateway-id "$igw" --vpc-id "$vpc" 2>/dev/null
+            aws ec2 delete-internet-gateway --internet-gateway-id "$igw" 2>/dev/null
+        done
+
+        # Delete subnets
+        local subnets=$(aws ec2 describe-subnets --filters "Name=vpc-id,Values=$vpc" --query 'Subnets[].SubnetId' --output text)
+        for subnet in $subnets; do
+            echo "   Deleting subnet: $subnet"
+            aws ec2 delete-subnet --subnet-id "$subnet" 2>/dev/null
+        done
+
+        # Delete route tables (except main)
+        local route_tables=$(aws ec2 describe-route-tables --filters "Name=vpc-id,Values=$vpc" --query 'RouteTables[?Associations[0].Main!=`true`].RouteTableId' --output text)
+        for rt in $route_tables; do
+            echo "   Deleting route table: $rt"
+            aws ec2 delete-route-table --route-table-id "$rt" 2>/dev/null
+        done
+
+        # Delete security groups (except default)
+        local security_groups=$(aws ec2 describe-security-groups --filters "Name=vpc-id,Values=$vpc" --query 'SecurityGroups[?GroupName!=`default`].GroupId' --output text)
+        for sg in $security_groups; do
+            echo "   Deleting security group: $sg"
+            aws ec2 delete-security-group --group-id "$sg" 2>/dev/null
+        done
+
+        # Delete VPC
+        echo "   Deleting VPC: $vpc"
+        aws ec2 delete-vpc --vpc-id "$vpc" 2>/dev/null
+        echo "✅ VPC $vpc and related resources deleted"
+    done
+}
+
+# Function to delete other tagged resources
+force_delete_other_resources() {
+    echo "🔍 Checking for other resources tagged with Workshop=true..."
+
+    # Delete Load Balancers
+    local elbs=$(aws elbv2 describe-load-balancers --query 'LoadBalancers[].LoadBalancerArn' --output text)
+    for elb_arn in $elbs; do
+        local tags=$(aws elbv2 describe-tags --resource-arns "$elb_arn" --query 'TagDescriptions[0].Tags[?Key==`Workshop`].Value' --output text 2>/dev/null)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Expressions don&#39;t expand in single quotes, use double quotes for that.